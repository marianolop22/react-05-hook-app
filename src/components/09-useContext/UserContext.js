import {createContext} from 'react';

//esto creaa un context
export const UserContext = createContext (null);