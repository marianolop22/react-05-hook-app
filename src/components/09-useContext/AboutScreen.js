import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const {user, setUser} = useContext(UserContext);

    const handleOnClick = (e) => {
        setUser({});
    };

    return (
        <div>
            <h1>About</h1>
            <hr/>
            <pre className="container">{JSON.stringify(user,null,3)}</pre>
            <button 
                className="btn btn-warning"
                onClick={handleOnClick}
            >Logout</button>
        </div>
    )
}
