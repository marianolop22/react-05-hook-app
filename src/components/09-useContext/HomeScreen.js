import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {

    //ac√° utilizo el hook enviando el context que tengo creado, me va a permitir laburar con eso
    const {user} = useContext(UserContext);

    console.log(user);


    return (
        <div>
            <h1>Home</h1>
            <hr/>
            <pre className="container">{JSON.stringify(user,null,3)}</pre>
        </div>
    )
}
