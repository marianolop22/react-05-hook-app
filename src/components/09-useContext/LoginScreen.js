import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    const loginUser = {
        id:1234,
        name: "Mariano"
    };

    return (
        <div>
            <h1>Login</h1>
            <hr/>
            <button 
                className="btn btn-primary"
                onClick={() => setUser(loginUser)}
            >Login</button>
        </div>
    )
}
