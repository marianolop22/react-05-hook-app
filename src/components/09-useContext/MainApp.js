
import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {

    const [user, setUser] = useState({});

    //Se declara el context mio, como padre de todo
    //Le mando como un objeto grande
    return (
        <UserContext.Provider value={{
            user,
            setUser
        }}>
            <AppRouter/>

        </UserContext.Provider>
    )
}
