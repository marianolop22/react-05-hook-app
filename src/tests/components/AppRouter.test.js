import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../components/09-useContext/AppRouter';
import { UserContext } from '../../components/09-useContext/UserContext';

describe ('Pruebas de AppRouter', ()=> {

    const user = {
        id:1,
        name:'Mariano'
    }
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter/>
        </UserContext.Provider>
    );

    test('debe de mostrar correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

});