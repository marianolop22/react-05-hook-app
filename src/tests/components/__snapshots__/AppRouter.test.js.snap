// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de AppRouter debe de mostrar correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      UseContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home
      </h1>
      <hr />
      <pre
        className="container"
      >
        {
   "id": 1,
   "name": "Mariano"
}
      </pre>
    </div>
  </div>
</div>
`;
